packages/website/plugins/generated-rule-docs.ts

```diff
@@ -0,0 +1,55 @@
+import type * as unist from 'unist';
+import type * as mdast from 'mdast';
+import type { Plugin } from 'unified';
+
+import * as eslintPlugin from '@typescript-eslint/eslint-plugin';
+
+const generatedRuleDocs: Plugin = () => {
+  return (root, file) => {
+    if (file.stem == null) {
+      return;
+    }
+
+    const docs = eslintPlugin.rules[file.stem]?.meta.docs;
+    if (!docs) {
+      return;
+    }
+
+    const parent = root as unist.Parent;
+
+    // 1. Add a description of the rule at the top of the file
+    parent.children.unshift({
+      children: [
+        {
+          children: [
+            {
+              type: 'text',
+              value: `${docs.description}.`,
+            },
+          ],
+          type: 'paragraph',
+        },
+      ],
+      type: 'blockquote',
+    } as mdast.Blockquote);
```

Comment:
Docusaurus adds an `h1` automatically if the file doesn't already start with one. So all we need to add is the description.

Reply:
1. Maybe use a paragraph instead of a quote?
2. Add this to the `RuleAttributes` component instead of in Markdown? The `description` is already part of the rule metadata available on client side.

Reply:
I like it as a quote though 😄: it adds subtle differentiation and matches the ESLint docs.

For `RuleAttributes`, that shows up after any existing initial paragraph 😕 so it would be out of order:

![Screenshot of the adjacent-overload-signatures rule page with the description quote after the general text paragraph](https://user-images.githubusercontent.com/3335181/180492753-bb336b7d-f2cb-4dfc-aaef-9adc0a466d56.png)